File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.py
class Solution(object):
    def equalPairs(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        n = len(grid)
        hashMap = defaultdict(int)

        for row in grid:
            rowStr = str(row)
            hashMap[rowStr] += 1
        
        res = 0
        for j in range(n):
            col = [grid[i][j] for i in range(n)]
            colStr = str(col)
            res += hashMap.get(colStr, 0)

        return res

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.go
func equalPairs(grid [][]int) int {
    n := len(grid)

    rows := make(map[string]int)
    for i := 0; i < n; i++ {
        key := fmt.Sprint(grid[i])
        rows[key]++
    }

    ans := 0
    for j := 0; j < n; j++ {
        col := make([]int, n)
        for i := 0; i < n; i++ {
            col[i] = grid[i][j]
        }

        ans += rows[fmt.Sprint(col)]
    }

    return ans
}
